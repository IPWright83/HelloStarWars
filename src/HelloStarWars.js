import React, { Component } from "react";

class HelloStarWars extends React.PureComponent {

	constructor(props) {
		super(props);
		this.state = {};
	}

	async componentDidMount() {
		const id = this.props.id;

		try {
			const response = await fetch(`https://swapi.co/api/people/${id}`);
			const { name } = await response.json();
			this.setState({ name });
		} catch (e) {
			this.setState({ name: "!dlroW... Oh dear, something went wrong" });
		}
	}

	render() {
		const name = this.state.name || "...";
		return(
			<h1>Hello {name}!</h1>
		);
	}
}

export default HelloStarWars;