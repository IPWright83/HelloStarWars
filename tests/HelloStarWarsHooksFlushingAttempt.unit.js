import React from "react";
import renderer from 'react-test-renderer';
import Adapter from "enzyme-adapter-react-16";
import enzyme from "enzyme";
import { mount } from "enzyme";
import { act } from 'react-dom/test-utils';

import HelloStarWars from "../src/HelloStarWarsHooks";

// using jest-fetch-mock

// Define a function which should allow any promises/macro tasks to complete
const flushMacroTasks = () => new Promise(resolve => setTimeout(resolve, 0));

enzyme.configure({ adapter: new Adapter() });

xdescribe("HelloStarWarsHooks Flushing Attempt", () => {

	it("Renders with a place holder", () => {
		const result = renderer
			.create(<HelloStarWars id="1" />)
			.toJSON();

		expect(result).toMatchSnapshot();
	});

	describe("ComponentDidMount", () => {

		beforeEach(() => {
			fetch.resetMocks();
		});

		it("Obtains result and sets data correctly", async () => {
			fetch.mockResponseOnce(JSON.stringify({ name: 'Jedi Master' }));

            const wrapper = await mount(<HelloStarWars id="1" />);
            const instance = wrapper.instance();

            // Wait for the react hook to complete
            await flushMacroTasks();

            wrapper.update();

            expect(wrapper).toMatchSnapshot();
		});

		xit("Handles an error in the web request correctly", async () => {
            fetch.mockReject(new Error("Invalid ID"));
            
            const wrapper = await mount(<HelloStarWars id="foobar" />);
			const instance = wrapper.instance();

            await flushMacroTasks();

			wrapper.update();
			expect(wrapper).toMatchSnapshot();
		});
	});
});