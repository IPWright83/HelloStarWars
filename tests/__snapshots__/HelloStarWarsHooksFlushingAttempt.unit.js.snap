// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloStarWarsHooks Flushing Attempt ComponentDidMount Obtains result and sets data correctly 1`] = `
<HelloStarWarsHooks
  id="1"
>
  <h1>
    Hello Jedi Master!
  </h1>
</HelloStarWarsHooks>
`;

exports[`HelloStarWarsHooks Flushing Attempt Renders with a place holder 1`] = `
<h1>
  Hello undefined!
</h1>
`;
